src/prefect/deployments/__init__.py

```diff
@@ -1,20 +1,33 @@
+from typing import TYPE_CHECKING
 from prefect._internal.compatibility.migration import getattr_migration
```

Comment:
Could it also be worthwhile to import this within `__getattr__`? Looks like this module imports from `prefect.exceptions` which has an `import prefect` statement.